<!-- This launch file launches a single mvBlueFox MLC usb 2.0 camera -->
<!-- You should provide the launch file with your camera serial number and
     calibration file -->

<launch>

  <!-- Serial number on the back of camera -->
  <arg name="serial" default="25000495" />
  <!-- Calibration files are located under bluefox/config -->
  <arg name="calibration_file" default="calib_$(arg serial).yml" />
  <!-- Options for mode are: master, slave and standalone -->
  <arg name="mode" default="standalone" />
  <arg name="fps" default="20.0" />
  <arg name="color" default="false" />
  <!-- White balance parameter, options are: indoor, outdoor -->
  <arg name="wbp" default="none" />

  <node pkg="bluefox2" type="bluefox2" name="cam_$(arg serial)" output="screen" >
    <param name="height" type="int" value="480" />
    <param name="width" type="int" value="752" />
    <param name="fps" type="double" value="$(arg fps)" />
    <param name="gain" type="double" value="0.0" />
    <param name="exposure_time_us" type="int" value="10000" />
    <param name="use_auto_exposure" type="bool" value="false" />
    <param name="use_color" type="bool" value="$(arg color)" />
    <param name="white_balance" type="string" value="$(arg wbp)" />
    <param name="use_binning" type="bool" value="false" />
    <param name="serial" type="string" value="$(arg serial)" />
    <param name="frame_id" type="string" value="cam$(arg serial)" />
    <param name="mode" type="string" value="$(arg mode)" />
    <param name="calibration_url" type="string" value="package://bluefox2/config/$(arg calibration_file)" />
  </node>

</launch>
